hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/HoodieSparkCopyOnWriteTable.java

```diff
@@ -210,12 +214,20 @@ public void rollbackBootstrap(HoodieEngineContext context, String instantTime) {
 
   protected HoodieMergeHandle getUpdateHandle(String instantTime, String partitionPath, String fileId,
       Map<String, HoodieRecord<T>> keyToNewRecords, HoodieBaseFile dataFileToBeMerged) {
+    Option<BaseKeyGenerator> keyGeneratorOpt = Option.empty();
+    if (!config.populateMetaFields()) {
+      try {
+        keyGeneratorOpt = Option.of((BaseKeyGenerator) HoodieSparkKeyGeneratorFactory.createKeyGenerator(new TypedProperties(config.getProps())));
+      } catch (IOException e) {
+        throw new HoodieIOException("Only BaseKyGenerators are supported when meta columns are disabled ", e);
```

Comment:
would a user understand `BaseKeyGenerators are supported` ? (note typo). Can we make exception messages more user-friendly?

Reply:
not sure what exactly we can say here. We added this new method, where you can fetch partition path from internal row for row writer path. So, key generators extending from BaseKeyGenerators only can be used when meta fields are disabled. Do you have any good suggestion here of what msg to convey.